src/cpp/rtps/transport/shared_mem/SharedMemManager.hpp

```diff
@@ -233,42 +420,57 @@ class SharedMemManager
                 uint32_t size,
                 const std::chrono::steady_clock::time_point& max_blocking_time_point)
         {
-            std::lock_guard<std::mutex> lock(alloc_mutex_);
+            (void)max_blocking_time_point;
 
-            wait_for_avaible_space(size, max_blocking_time_point);
+            std::lock_guard<std::mutex> lock(alloc_mutex_);
 
-            release_unused_buffers();
+            if (!recover_buffers(size))
```

Comment:
Should we always recover buffers?

I believe we could skip this if there are enough `free_bytes_` and there is at least on entry on `free_buffers_`

Reply:
We've talked about this.

Reply:
We'll leave improvements here for a separate PR